### ISSUE TITLE
Will IIC work for small dataset?

### ISSUE THREAD

**Issue by primecai** [xu-ji/IIC#53]
Hello,

I have a small dataset of around 2000 images, 10 classes(approximately 200 per class). I am trying to use the settings for CIFAR10 to train IIC on this dataset, but after 50+ epochs it is still stucking at around 16% accuracy, which is pretty much the same as after the first epoch. I'm wondering if it is because of dataset size?

Many thanks.

**Comment by xu-ji** [comment]
Larger datasets (specifically, more samples per class) produce better accuracy in general. This is even more true in unsupervised settings where the training signal is not so clear.
However, for simple tasks 2k images may be enough. There is no way to know for sure (for either increasing max accuracy or decreasing convergence speed, as you mention) unless you are able try several differently sized variants of your dataset. High convergence speed is not really an issue, actually, max accuracy is ultimately what matters.

You should definitely try to change the learning rate and other hyperparameters (see table 2 in [supp mat](https://github.com/xu-ji/IIC/blob/master/paper/suppmat_camera_ready.pdf) for all the ones we used).

Since the objective is unsupervised and unlabelled data is cheap, another thing to try would be to bring in other data from some other source and train on both, similar to what we did with STL10, which is only 5k "main" images + 100k other images, 5k size comparable to your 2k.
